editor/effects.js

```diff
@@ -20,9 +20,30 @@ import {
 	getCurrentPostType,
 	getBlocks,
 	getPostEdits,
+	isCleanNewPost,
 } from './selectors';
 
+let originalDocumentTitle;
+
+function populateDocumentTitle( title, isCleanNew = false ) {
+	if ( ! title.trim() ) {
```

Comment:
Seems like some of this logic makes more sense in a selector, like `getDocumentTitle`.